perl/imap/IMAP.xs

```diff
@@ -717,26 +717,23 @@ imclient_toURL(client,server,box)
         char *server
         char *box
 PREINIT:
-        char *out_buf;
-        int len;
+        struct buf buf = BUF_INITIALIZER;
         struct imapurl imapurl;
 PPCODE:
-        len = strlen(server)+strlen(box);
-        out_buf = safemalloc(4*len);
-
         memset(&imapurl, 0, sizeof(struct imapurl));
         imapurl.server = server;
         imapurl.mailbox = box;
-        imapurl_toURL(out_buf, &imapurl);
+        imapurl_toURL(&buf, &imapurl);
+        buf_cstring(&buf);
 
-        if(!out_buf[0]) {
-                safefree(out_buf);
+        if(!buf_len(&buf)) {
+                buf_free(&buf);
                 XSRETURN_UNDEF;
         }
 
-        XPUSHs(sv_2mortal(newSVpv(out_buf, 0)));
+        XPUSHs(sv_2mortal(newSVpv(buf_release(&buf), 0)));
```

Comment:
I'm not sure if this is correct, but I'm also not sure how to find out.  Maybe Rik or Rob M might know?

The original `out_buf` was managed with `safemalloc()`/`safefree()`, but the new one in `buf` will be from ordinary `malloc()` or `realloc()`.  `newSVpv()` looks like it might care about the difference... I'm guessing its name might stand for something like "new safe value from pointer value", in which case it might not be valid to pass it memory from an ordinary malloc.

Reply:
right, newSVpv takes a const pointer. I've changed buf_release to buf_cstring here. thanks.